app/org/sagebionetworks/bridge/services/ConsentServiceImpl.java

```
@@ -67,49 +71,61 @@ public void setApplicationEventPublisher(ApplicationEventPublisher publisher) {
     public ConsentSignature getConsentSignature(final User caller, final Study study) {
         checkNotNull(caller, Validate.CANNOT_BE_NULL, "user");
         checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
-
         final StudyConsent consent = studyConsentDao.getConsent(study.getIdentifier());
         if (consent == null) {
             throw new EntityNotFoundException(StudyConsent.class);
         }
-        ConsentSignature consentSignature = userConsentDao.getConsentSignature(caller.getHealthCode(), consent);
+        Account account = stormpathClient.getResource(caller.getStormpathHref(), Account.class);
+        ConsentSignature consentSignature = accountEncryptionService.getConsentSignature(study, account);
```

Could stormpath client or aes throw? If so, should we propagate the error? Or should we log the exception and fall back to dynamo?